### ISSUE TITLE
DeviceGroup does not work with IPv6

### ISSUE THREAD

**Issue by tkam8** [F5Networks/f5-declarative-onboarding#233]
### Environment
 * Declarative Onboarding Version: 1.20
 * BIG-IP Version: 15.1.2.1  0.408.10

### Summary

When using DO to create a cluster between two BIG-IP configured using completely IPv6 configsync and management IPs, device group fails to be configured. This is similar to https://github.com/F5Networks/f5-declarative-onboarding/issues/199  but I am not configuring any IPv4 addresses. 

This is reproduceable. 

### Steps To Reproduce
1. Submit the following declaration:

BIGIP1
```json
{
    "$schema": "https://raw.githubusercontent.com/F5Networks/f5-declarative-onboarding/master/src/schema/latest/base.schema.json",
    "schemaVersion": "1.20.0",
    "class": "Device",
    "async": true,
    "label": "Onboard BIG-IP into DSC cluster",
    "Common": {
        "class": "Tenant",
        "myProvisioning": {
            "class": "Provision",
            "afm": "none",
            "gtm": "none",
            "ltm": "nominal",
            "pem": "none"
        },
        "trunk_core": {
            "class": "Trunk",
            "interfaces": [
                "1.3",
                "1.4"
            ]
        },
        "trunk_mgmt": {
            "class": "Trunk",
            "interfaces": [
                "1.5",
                "1.6"
            ]
        },
        "vl_core_ex_v6": {
            "class": "VLAN",
            "cmpHash": "src-ip",
            "tag": 1002,
            "interfaces": [
                {
                    "name": "trunk_core",
                    "tagged": false
                }
            ]
        },
        "vl_mgmt_ex_v6": {
            "class": "VLAN",
            "cmpHash": "src-ip",
            "tag": 1001,
            "interfaces": [
                {
                    "name": "trunk_mgmt",
                    "tagged": false
                }
            ]
        },
        "vl_ov_ha_int_v6": {
            "class": "VLAN",
            "cmpHash": "src-ip",
            "tag": 1000,
            "interfaces": [
                {
                    "name": "1.1",
                    "tagged": false
                }
            ]
        },
        "ip_core_a_ex_v6_nonfl": {
            "class": "SelfIp",
            "address": "fdc3:eaf2:d8b9:a:0000:0000:0000:0101/64",
            "vlan": "vl_core_ex_v6",
            "allowService": "default",
            "trafficGroup": "traffic-group-local-only"
        },
        "ip_core_a_ex_v6_fl": {
            "class": "SelfIp",
            "address": "fdc3:eaf2:d8b9:a:0000:0000:0000:0100/64",
            "vlan": "vl_core_ex_v6",
            "allowService": "default",
            "trafficGroup": "traffic-group-1"
        },
        "ip_mgmt_a_ex_v6_nonfl": {
            "class": "SelfIp",
            "address": "fdc3:eaf2:d8b9:b:0000:0000:0000:0101/64",
            "vlan": "vl_mgmt_ex_v6",
            "allowService": "default",
            "trafficGroup": "traffic-group-local-only"
        },
        "ip_mgmt_a_ex_v6_fl": {
            "class": "SelfIp",
            "address": "fdc3:eaf2:d8b9:b:0000:0000:0000:0100/64",
            "vlan": "vl_mgmt_ex_v6",
            "allowService": "default",
            "trafficGroup": "traffic-group-1"
        },
        "ip_ha_int_v6_nonfl": {
            "class": "SelfIp",
            "address": "fdc3:eaf2:d8b9:123a:0000:0000:0000:0001/64",
            "vlan": "vl_ov_ha_int_v6",
            "allowService": "default",
            "trafficGroup": "traffic-group-local-only"
        },
        "dbvars": {
            "class": "DbVariables",
            "tm.tcpsegmentationoffload": "disable",
            "tm.tcplargereceiveoffload": "disable",
            "tm.tcpwoodsidefasterrecovery": "enable",
            "tmm.cec.classifier.abr.enable": true,
            "ui.system.preferences.recordsperscreen": 100,
            "iprep.autoupdate": "disable",
            "provision.extramb": "1000",
            "restjavad.useextramb": "true",
            "setup.run": "false"
        },
        "mySystem": {
            "class": "System",
            "hostname": "bigip1.local",
            "cliInactivityTimeout": 900,
            "consoleInactivityTimeout": 900
        },
        "configsync": {
            "class": "ConfigSync",
            "configsyncIp": "/Common/ip_ha_int_v6_nonfl/address"
        },
        "failoverAddress": {
            "class": "FailoverUnicast",
            "address": "/Common/ip_ha_int_v6_nonfl/address"
        },
        "failoverGroup": {
            "class": "DeviceGroup",
            "type": "sync-failover",
            "members": [ "fdc3:eaf2:d8b9:284c:0000:0000:0000:0001", "fdc3:eaf2:d8b9:284c:0000:0000:0000:0002"],
            "owner": "fdc3:eaf2:d8b9:284c:0000:0000:0000:0001",
            "autoSync": true,
            "saveOnAutoSync": false,
            "networkFailover": true,
            "asmSync": false
        },
        "trust": {
            "class": "DeviceTrust",
            "localUsername": "admin",
            "localPassword": "passwd",
            "remoteHost": "/Common/failoverGroup/members/0",
            "remoteUsername": "admin",
            "remotePassword": "passwd"
        }
    }
}
```

BIGIP2
```json
{
    "$schema": "https://raw.githubusercontent.com/F5Networks/f5-declarative-onboarding/master/src/schema/latest/base.schema.json",
    "schemaVersion": "1.20.0",
    "class": "Device",
    "async": true,
    "label": "Onboard BIG-IP into DSC cluster",
    "Common": {
        "class": "Tenant",
        "myProvisioning": {
            "class": "Provision",
            "afm": "none",
            "gtm": "none",
            "ltm": "nominal",
            "pem": "none"
        },
        "trunk_core": {
            "class": "Trunk",
            "interfaces": [
                "1.3",
                "1.4"
            ]
        },
        "trunk_mgmt": {
            "class": "Trunk",
            "interfaces": [
                "1.5",
                "1.6"
            ]
        },
        "vl_core_ex_v6": {
            "class": "VLAN",
            "cmpHash": "src-ip",
            "tag": 1002,
            "interfaces": [
                {
                    "name": "trunk_core",
                    "tagged": false
                }
            ]
        },
        "vl_mgmt_ex_v6": {
            "class": "VLAN",
            "cmpHash": "src-ip",
            "tag": 1001,
            "interfaces": [
                {
                    "name": "trunk_mgmt",
                    "tagged": false
                }
            ]
        },
        "vl_ov_ha_int_v6": {
            "class": "VLAN",
            "cmpHash": "src-ip",
            "tag": 1000,
            "interfaces": [
                {
                    "name": "1.1",
                    "tagged": false
                }
            ]
        },
        "ip_core_a_ex_v6_nonfl": {
            "class": "SelfIp",
            "address": "fdc3:eaf2:d8b9:a:0000:0000:0000:0102/64",
            "vlan": "vl_core_ex_v6",
            "allowService": "default",
            "trafficGroup": "traffic-group-local-only"
        },
        "ip_mgmt_a_ex_v6_nonfl": {
            "class": "SelfIp",
            "address": "fdc3:eaf2:d8b9:b:0000:0000:0000:0102/64",
            "vlan": "vl_mgmt_ex_v6",
            "allowService": "default",
            "trafficGroup": "traffic-group-local-only"
        },
        "ip_ha_int_v6_nonfl": {
            "class": "SelfIp",
            "address": "fdc3:eaf2:d8b9:123a:0000:0000:0000:0002/64",
            "vlan": "vl_ov_ha_int_v6",
            "allowService": "default",
            "trafficGroup": "traffic-group-local-only"
        },
        "dbvars": {
            "class": "DbVariables",
            "tm.tcpsegmentationoffload": "disable",
            "tm.tcplargereceiveoffload": "disable",
            "tm.tcpwoodsidefasterrecovery": "enable",
            "tmm.cec.classifier.abr.enable": true,
            "ui.system.preferences.recordsperscreen": 100,
            "iprep.autoupdate": "disable",
            "provision.extramb": "1000",
            "restjavad.useextramb": "true",
            "setup.run": "false"
        },
        "mySystem": {
            "class": "System",
            "hostname": "bigip2.local",
            "cliInactivityTimeout": 900,
            "consoleInactivityTimeout": 900
        },
        "configsync": {
            "class": "ConfigSync",
            "configsyncIp": "/Common/ip_ha_int_v6_nonfl/address"
        },
        "failoverAddress": {
            "class": "FailoverUnicast",
            "address": "/Common/ip_ha_int_v6_nonfl/address"
        },
        "failoverGroup": {
            "class": "DeviceGroup",
            "type": "sync-failover",
            "members": [ "fdc3:eaf2:d8b9:284c:0000:0000:0000:0001", "fdc3:eaf2:d8b9:284c:0000:0000:0000:0002"],
            "owner": "fdc3:eaf2:d8b9:284c:0000:0000:0000:0001",
            "autoSync": true,
            "saveOnAutoSync": false,
            "networkFailover": true,
            "asmSync": false
        },
        "trust": {
            "class": "DeviceTrust",
            "localUsername": "admin",
            "localPassword": "passwd",
            "remoteHost": "/Common/failoverGroup/members/0",
            "remoteUsername": "admin",
            "remotePassword": "passwd"
        }
    }
}

```

### Expected Behavior
Cluster is created

### Actual Behavior
Declaration is accepted 202, selfip and vlans get created but device group and trust never gets created, eventually rollback. 

 /var/log/restnoded/restnoded.log
```
Thu, 13 May 2021 14:28:43 GMT - finest: [f5-declarative-onboarding: restWorker.js] tryUntil: got error {"code":404}
Thu, 13 May 2021 14:28:43 GMT - finest: [f5-declarative-onboarding: restWorker.js] typeof err object
Thu, 13 May 2021 14:28:43 GMT - finer: [f5-declarative-onboarding: restWorker.js] tryUntil error: 01020036:3: The requested device group (/Common/failoverGroup) was not found. tries left: 0
Thu, 13 May 2021 14:28:43 GMT - finest: [f5-declarative-onboarding: restWorker.js] tryUntil: retryOrReject: numRemaining: 0 , code: 404 , message: 01020036:3: The requested device group (/Common/failoverGroup) was not found.
Thu, 13 May 2021 14:28:43 GMT - finer: [f5-declarative-onboarding: restWorker.js] Max tries reached.
Thu, 13 May 2021 14:28:43 GMT - finest: [f5-declarative-onboarding: restWorker.js] tryUntil: got error {"code":404,"message":"tryUntil: max tries reached: 01020036:3: The requested device group (/Common/failoverGroup) was not found.","name":"Error"}
Thu, 13 May 2021 14:28:43 GMT - finest: [f5-declarative-onboarding: restWorker.js] typeof err object
Thu, 13 May 2021 14:28:43 GMT - finer: [f5-declarative-onboarding: restWorker.js] tryUntil error: tryUntil: max tries reached: 01020036:3: The requested device group (/Common/failoverGroup) was not found. tries left: 227
Thu, 13 May 2021 14:28:43 GMT - finest: [f5-declarative-onboarding: restWorker.js] tryUntil: retryOrReject: numRemaining: 227 , code: 404 , message: tryUntil: max tries reached: 01020036:3: The requested device group (/Common/failoverGroup) was not found.
Thu, 13 May 2021 14:28:53 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /shared/echo/available
Thu, 13 May 2021 14:28:53 GMT - fine: [f5-declarative-onboarding: restWorker.js] {}
Thu, 13 May 2021 14:28:53 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /shared/identified-devices/config/device-info/available
Thu, 13 May 2021 14:28:53 GMT - fine: [f5-declarative-onboarding: restWorker.js] {}
Thu, 13 May 2021 14:28:53 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /tm/sys/available
Thu, 13 May 2021 14:28:53 GMT - fine: [f5-declarative-onboarding: restWorker.js] {}
Thu, 13 May 2021 14:28:53 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /tm/cm/available
Thu, 13 May 2021 14:28:53 GMT - fine: [f5-declarative-onboarding: restWorker.js] {}
Thu, 13 May 2021 14:28:53 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /shared/iapp/package-management-tasks/available
Thu, 13 May 2021 14:28:53 GMT - fine: [f5-declarative-onboarding: restWorker.js] {}
Thu, 13 May 2021 14:28:53 GMT - fine: [f5-declarative-onboarding: restWorker.js] list localhost /tm/sys/mcp-state
```

confirmed that 

> The remoteHost property of the DeviceTrust class should be the same device as the owner of the Device Group.
https://clouddocs.f5.com/products/extensions/f5-declarative-onboarding/latest/clustering-managing-devices.html

**Comment by tkam8** [comment]
internal case: C3574313

**Comment by enceladus-beep** [comment]
> 
> 
> internal case: C3574313

Working on a repro of this problem. 

There is an error in the templates (123a vs 284c)

```
[...]
        "ip_ha_int_v6_nonfl": {
            "class": "SelfIp",
            "address": "fdc3:eaf2:d8b9:123a:0000:0000:0000:0002/64",
[...]
       "failoverGroup": {
            "class": "DeviceGroup",
            "type": "sync-failover",
            "members": [ "fdc3:eaf2:d8b9:284c:0000:0000:0000:0001", "fdc3:eaf2:d8b9:284c:0000:0000:0000:0002"],
            "owner": "fdc3:eaf2:d8b9:284c:0000:0000:0000:0001",
            "autoSync": true,
[...]
```

But I was still able to reproduce the problem after correcting the IPv6 addresses 

```
Fri, 14 May 2021 13:19:38 GMT - finer: [f5-declarative-onboarding: restWorker.js] tryUntil error: tryUntil: max tries reached: 01020036:3: The requested device group (/Common/failoverGroup) was not found. tries left: 224
Fri, 14 May 2021 13:19:38 GMT - finest: [f5-declarative-onboarding: restWorker.js] tryUntil: retryOrReject: numRemaining: 224 , code: 404 , message: tryUntil: max tries reached: 01020036:3: The requested device group (/Common/failoverGroup) was not found.
```

**Comment by dstokesf5** [comment]
Thank you for your feedback. I have added this issue to our internal product backlog as AUTOTOOL-2502.

**Comment by tkam8** [comment]
Confirmed deviceGroup was created when using latest build 3 of v1.22.0

thanks all

**Comment by dstokesf5** [comment]
This issue has been resolved by the 1.22.0 release.
